### ISSUE TITLE
0.9 compat

### ISSUE THREAD

**Issue by chucker** [chanan/BlazorStrap#42]
* `JSInterop.Current` is gone; we need to use injection instead.
* [EventCallback](https://github.com/aspnet/AspNetCore/issues/6351) also appears to break `BlazorButton`.

**Comment by chanan** [comment]
I haven't look at it yet, but I fear the new jsruntime injection might be problematic, hopefully not...

**Comment by chucker** [comment]
>I fear the new jsruntime injection might be problematic

Possibly (since `BlazorStrapInterop` methods are currently static).

I've tried testing it, but now instead run into problems with `BlazorButton`, which appears to be because of a change to how `DynamicElement` behaves…

**Comment by chanan** [comment]
Oh no... I use the same idea in another project quite extensively... ugh. Well, I now know what my weekend is going to be filled with.

**Comment by chucker** [comment]
Yup, I've narrowed _that_ problem down to `DynamicElement`. If I leave that out of `BlazorButton`, I can get `onclick` working again.

And it looks like my JSInterop approach is fixed.

…but it also appears popover positioning is now broken. Will need to investigate that separately. And of course I broke the entire DynamicElement stuff.

**Comment by chanan** [comment]
Sounds like fun...

**Comment by chucker** [comment]
So the problem with `BlazorButton` (and possibly other similar stuff) is how the `onclick` attribute gets rendered.

On the browser end, I now see

```html
<button onclick="Microsoft.AspNetCore.Components.EventCallback`1[[Microsoft.AspNetCore.Components.UIMouseEventArgs, Microsoft.AspNetCore.Components, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]]" type="button" class="btn btn-primary      " role="button" id="popover4" _bl_125="">
```

Whereas before, the `onclick` attribute was missing from the DOM entirely. It appears to be an artifact of the move to `EventCallback` (instead of `Action`), and I'm not convinced this isn't also a bug on Blazor's end.

In the generated `.cs`, we used to see:

```csharp
                builder3.AddAttribute(105, "onclick", new System.Action<Microsoft.AspNetCore.Components.UIMouseEventArgs>(ToggleThingy));
```

This has changed to:

```csharp
                builder3.AddAttribute(105, "onclick", Microsoft.AspNetCore.Components.RuntimeHelpers.TypeCheck<Microsoft.AspNetCore.Components.EventCallback<Microsoft.AspNetCore.Components.UIMouseEventArgs>>(Microsoft.AspNetCore.Components.EventCallback.Factory.Create<Microsoft.AspNetCore.Components.UIMouseEventArgs>(ToggleThingy)));
```

My guess is TypeCheck is intended for serializing the attribute content to HTML, and that it shouldn't be doing that at all.

**Comment by chucker** [comment]
> it also appears popover positioning is now broken.

This might be just a side effect of `DynamicElement`, actually.

**Comment by chucker** [comment]
Yup, popovers are fine.

Really quick and dirty fix to get `BlazorButton` working again:

```csharp
            foreach (var param in _attributesToRender)
            {
                switch (param.Value)
                {
                    case EventCallback<UIMouseEventArgs> ec:
                        builder.AddAttribute<UIMouseEventArgs>(1, param.Key, ec);

                        break;
                    default:
                        builder.AddAttribute(1, param.Key, param.Value);

                        break;
                }
            }
```

I.e., there seems to be an overload resolution error here.

**Comment by chanan** [comment]
That is awesome! Thanks for figuring that out!

**Comment by chucker** [comment]
(I think I'll go file a bug with AspNetCore and see if they agree. I think they broke something about the overload resolution.)

**Comment by chanan** [comment]
merged #43
